mewarpx/tests/test_custom_criteria.py

```diff
@@ -1,18 +1,10 @@
-"""Test the custom criteria using capacitive discharge"""
+"""Test the custom criteria to terminate a simulation and the
+functionality to add new pid's at runtime."""
```

Comment:
I would prefer these were two separate tests in two separate files. It's better to clearly understand what is tested in each test file, and have each test be the one to fail when that's the specific functionality that broke.

Reply:
In general I agree with that for heavy functionality. My thinking here is that the custom criteria test is testing very robust functionality (that a function which returns False could pass that value to another function). In reality it seems that the custom criteria test would more likely fail if something goes wrong in the function to collect particle data (total number in this case), so it seemed to make sense to wrap the test up with another test that checks the particle data collection bridge.